operators/derive/to_es.py

```diff
@@ -1,38 +1,67 @@
 import dataflows as DF
-from dataflows.helpers.resource_matcher import ResourceMatcher
 from dataflows_elasticsearch import dump_to_es
+from tableschema_elasticsearch.mappers import MappingGenerator
 
 from conf import settings
 from srm_tools.logger import logger
 
 
-def set_es_item_types(resources=None):
-    def func(package):
-        matcher = ResourceMatcher(resources, package.pkg)
-        for resource in package.pkg.descriptor['resources']:
-            if matcher.match(resource['name']):
-                for field in resource["schema"]["fields"]:
-                    if field["type"] == "any":
-                        field["type"] = "string"
-                    elif field["type"] == "array":
-                        field["es:itemType"] = "string"
-        yield package.pkg
-
-        res_iter = iter(package)
-        for r in res_iter:
-            if matcher.match(r.res.name):
-                yield r.it
-            else:
-                yield r
-
-    return func
+class SRMMappingGenerator(MappingGenerator):
+    @classmethod
+    def _convert_type(cls, schema_type, field, prefix):
+        # TODO: should be in base class
+        if field['type'] == 'any':
+            field['es:itemType'] = 'string'
+        prop = super()._convert_type(schema_type, field, prefix)
+        boost, keyword = field.get('es:boost'), field.get('es:keyword')
+        if keyword:
+            prop['type'] = 'keyword'
+        if boost:
+            prop['boost'] = boost
+        if schema_type in ('number', 'integer', 'geopoint'):
+            prop['index'] = True
+        return prop
 
 
 def data_api_es_flow():
     return DF.Flow(
-        DF.load(f'{settings.DATA_DUMP_DIR}/table_data/datapackage.json'),
+        DF.load(f'{settings.DATA_DUMP_DIR}/card_data/datapackage.json'),
+        DF.set_type('card_id', type='string', **{'es:keyword': True}),
```

Comment:
there's no need to re-set the type here, if it's already set.
i.e. `DF.set_type('card_id',, **{'es:keyword': True}),` is also valid

Reply:
Ok.